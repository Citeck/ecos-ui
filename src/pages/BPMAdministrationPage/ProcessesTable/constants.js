import { BpmnProcessStatusColumn } from '../../../components/BpmnProcessStatusColumn';
import { URL } from '../../../constants';
import { getMLValue } from '../../../helpers/util';

export const PROCESSED_TABLE_COLUMNS = [
  {
    newFormatter: { type: 'link', config: { getUrl: row => `${URL.BPMN_ADMIN_PROCESS}?recordRef=${row.id}` } },
    text: getMLValue({
      ru: 'ID Процесса',
      en: 'Process ID'
    }),
    attribute: 'key',
    dataField: 'key'
  },
  {
    text: getMLValue({
      ru: 'Описание процесса',
      en: 'Process definition'
    }),
    attribute: 'ecosDef',
    dataField: 'ecosDef'
  },
  {
    text: getMLValue({
      ru: 'Количество инцидентов',
      en: 'Incident count'
    }),
    attribute: 'incidentCount',
    dataField: 'incidentCount'
  },
  {
    text: getMLValue({
      ru: 'Количество запущенных экземпляров',
      en: 'Running process count'
    }),
    attribute: 'instancesCount',
    dataField: 'instancesCount'
  },
  {
    classes: 'common-table-column__small',
    newFormatter: { type: 'component', config: { Component: BpmnProcessStatusColumn } },
    text: getMLValue({
      ru: 'Состояние',
      en: 'State'
    }),
    attribute: 'key',
    dataField: 'key'
  }
];
