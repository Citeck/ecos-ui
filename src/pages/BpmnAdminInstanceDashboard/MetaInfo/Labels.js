export default {
  WIDGET_TITLE: 'instance-admin.info-widget.title',

  ID: 'instance-admin.info-widget.id',
  VERSION: 'instance-admin.info-widget.version',
  DOCUMENT: 'instance-admin.info-widget.document',
  CHOUSE_ACTION: 'instance-admin.info-widget.choose-action',
  DEFINITION_REF: 'instance-admin.info-widget.definition-ref',
  START_TIME: 'instance-admin.info-widget.start-time',
  PAUSED: 'instance-admin.info-widget.paused',

  NOT_DEFINED: 'instance-admin.info-widget.not-defined',

  ACTIONS_TITLE: 'instance-admin.info-widget.actions-title',

  TOKENS_DROPDOWN_TITLE: 'migration-modal.dropdown-title',
  VERSIONS_DROPDOWN_TITLE: 'migration-versions.dropdown-title'
};
