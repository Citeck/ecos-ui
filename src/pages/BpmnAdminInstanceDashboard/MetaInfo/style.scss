@import '../../../styles/constants.scss';

.bpmn-admin-info {
  @extend %column-wrapper-16;

  padding: 16px;

  &__main {
    display: flex;
    justify-content: space-between;

    &-item {
      @extend %column-wrapper-8;

      width: 300px;

      & .ecos-btn__content {
        display: flex;
        justify-content: space-between;
      }
    }
  }

  &__dropdown-link {
    color: initial;

    &:hover {
      color: initial;
    }
  }

  &__actions {
    width: 300px;
  }

  &__secondary {
    @extend %row-wrapper-8;

    display: flex;
    justify-content: space-between;

    color: $color-gray-5;

    font-size: medium;
  }

  &__pause-icon {
    cursor: pointer;
    color: $color-blue-light;
  }

  &__buttons {
    @extend %column-wrapper-8;

    width: 250px;

    &-wrapper {
      display: flex;
    }
  }
}
