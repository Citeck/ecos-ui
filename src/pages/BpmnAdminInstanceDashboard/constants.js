export const INSTANCE_ADMIN_BLOCK_CLASS = 'bpmn-admin-instance';

export const META_INFO_BLOCK_CLASS = 'bpmn-admin-info';
export const JOURNALS_TABS_BLOCK_CLASS = 'bpmn-admin-tabs';

export const VARIABLE_VALUE_BLOCK_CLASS = 'variable-value-modal';

export const MIGRATION_MODAL_BLOCK_CLASS = 'migration-modal';
