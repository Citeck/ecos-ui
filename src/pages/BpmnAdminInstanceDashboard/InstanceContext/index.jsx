import React, { useState } from 'react';

import { getSearchParams } from '../../../helpers/util';
import { INSTANCE_TABS_TYPES } from '../../../constants/instanceAdmin';

export const InstanceContext = React.createContext();

export const InstanceContextProvider = props => {
  const urlParams = getSearchParams();
  const { recordRef } = urlParams;

  const [isSuspended, setIsSuspended] = useState(null);
  const [activeTabId, setActiveTabId] = useState(INSTANCE_TABS_TYPES.VARIABLES);

  const [, dispInstanceId] = (recordRef || '').split('@');

  const [activityElement, setActivityElement] = useState(null);

  return (
    <InstanceContext.Provider
      value={{
        instanceId: recordRef,
        dispInstanceId,

        isSuspended,
        setIsSuspended,

        activityElement,
        setActivityElement,

        activeTabId,
        setActiveTabId
      }}
    >
      {props.children}
    </InstanceContext.Provider>
  );
};
