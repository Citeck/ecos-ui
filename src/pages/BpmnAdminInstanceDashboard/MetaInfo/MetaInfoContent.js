import React, { useContext, useEffect } from 'react';
import moment from 'moment';
import { connect } from 'react-redux';

import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';

import { selectInstanceMetaInfo } from '../../../selectors/instanceAdmin';
import PanelTitle, { COLOR_GRAY } from '../../../components/common/PanelTitle/PanelTitle';
import { Btn } from '../../../components/common/btns';
import PageService from '../../../services/PageService';
import { createDocumentUrl } from '../../../helpers/urls';
import { getMetaInfo } from '../../../actions/instanceAdmin';
import { t } from '../../../helpers/util';
import { InstanceContext } from '../InstanceContext';
import { META_INFO_BLOCK_CLASS } from '../constants';
import Labels from './Labels';
import { ActionsButton } from './components';

const MetaInfoContent = ({ getMetaInfo, metaInfo }) => {
  const { instanceId, setIsSuspended } = useContext(InstanceContext);
  useEffect(
    () => {
      if (isEmpty(metaInfo)) {
        isFunction(getMetaInfo) && getMetaInfo(instanceId);
      }
    },
    [instanceId]
  );

  useEffect(
    () => {
      if (metaInfo) {
        setIsSuspended(metaInfo.isSuspended);
      }
    },
    [metaInfo]
  );

  const handleLinkClick = href => {
    PageService.changeUrlLink(createDocumentUrl(href), {
      openNewTab: true
    });
  };

  return (
    <div className={META_INFO_BLOCK_CLASS}>
      <div className={`${META_INFO_BLOCK_CLASS}__main`}>
        <div className={`${META_INFO_BLOCK_CLASS}__main-item`}>
          <PanelTitle narrow color={COLOR_GRAY}>
            {t(Labels.DOCUMENT)}
          </PanelTitle>
          <Btn className="ecos-btn_blue" onClick={() => handleLinkClick(metaInfo.documentId)} disabled={!metaInfo.document}>
            {metaInfo.document || t(Labels.NOT_DEFINED)}
            <i className="icon-small-right" />
          </Btn>
        </div>

        <ActionsButton instanceId={instanceId} />

        <div className={`${META_INFO_BLOCK_CLASS}__main-item`}>
          <PanelTitle narrow color={COLOR_GRAY}>
            {t(Labels.DEFINITION_REF)}
          </PanelTitle>
          <Btn className="ecos-btn_blue" onClick={() => handleLinkClick(metaInfo.definitionRefId)} disabled={!metaInfo.definitionRef}>
            {metaInfo.definitionRef || t(Labels.NOT_DEFINED)}
            <i className="icon-small-right" />
          </Btn>
        </div>
      </div>

      <div className={`${META_INFO_BLOCK_CLASS}__secondary`}>
        <span>{`${t(Labels.VERSION)}: ${metaInfo.version || t(Labels.NOT_DEFINED)}`}</span>
        <span>{`${t(Labels.START_TIME)}: ${metaInfo.startTime ? moment(metaInfo.startTime).format('LLL') : t(Labels.NOT_DEFINED)}`}</span>
      </div>
    </div>
  );
};

const mapStateToProps = (store, props) => ({
  metaInfo: selectInstanceMetaInfo(store, props)
});

const mapDispatchToProps = dispatch => ({
  getMetaInfo: instanceId => dispatch(getMetaInfo({ instanceId }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MetaInfoContent);
