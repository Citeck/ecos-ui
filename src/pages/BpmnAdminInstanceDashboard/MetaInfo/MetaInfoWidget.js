import React from 'react';

import BaseWidget from '../../../components/widgets/BaseWidget';
import Dashlet from '../../../components/Dashlet';
import { t } from '../../../helpers/util';
import MetaInfoContent from './MetaInfoContent';
import Labels from './Labels';

import './style.scss';

class MetaInfoWidget extends BaseWidget {
  render() {
    const { instanceId } = this.props;

    return (
      <Dashlet
        title={t(Labels.WIDGET_TITLE)}
        needGoTo={false}
        onToggleCollapse={this.handleToggleContent}
        isCollapsed={this.isCollapsed}
        setRef={this.setDashletRef}
      >
        <MetaInfoContent instanceId={instanceId} />
      </Dashlet>
    );
  }
}

export default MetaInfoWidget;
