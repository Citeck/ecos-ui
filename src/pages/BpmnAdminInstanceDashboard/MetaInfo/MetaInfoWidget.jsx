import React from 'react';

import BaseWidget from '../../../components/widgets/BaseWidget';
import Dashlet from '../../../components/Dashlet';
import { t } from '../../../helpers/util';
import MetaInfoContent from './MetaInfoContent';
import Labels from './Labels';

import './style.scss';

class MetaInfoWidget extends BaseWidget {
  render() {
    const { instanceId } = this.props;

    return (
      <Dashlet title={t(Labels.WIDGET_TITLE)} setRef={this.setDashletRef} needGoTo={false} disableCollapse>
        <MetaInfoContent instanceId={instanceId} />
      </Dashlet>
    );
  }
}

export default MetaInfoWidget;
