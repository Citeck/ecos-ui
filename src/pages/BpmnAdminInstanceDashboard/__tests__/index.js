import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'enzyme';
import configureStore from '../../../store';
import {
  getCalledProcessesColumns,
  getExternalTasksColumns,
  getIncidentsColumns,
  getJobDefinitionsColumns,
  getVariableColumns
} from '../JournalsTabs/columns';
import { INSTANCE_TABS_TYPES } from '../../../constants/instanceAdmin';
import { InstanceContextProvider } from '../InstanceContext';
import MetaInfo from '../MetaInfo';

const initialState = {};

const tableProps = {
  instanceId: 'test-instance-id',
  actions: []
};

const renderWithRedux = Component => {
  const store = configureStore(null, { bpmnAdmin: initialState });

  return (
    <Provider store={store}>
      <InstanceContextProvider>
        <Component />
      </InstanceContextProvider>
    </Provider>
  );
};
describe('', () => {
  describe('', () => {
    it('should render MetaInfo component', () => {
      const component = render(renderWithRedux(() => <MetaInfo instanceId="test-instance-id" />));
      expect(component).toMatchSnapshot();
    });

    it('should render corrent columns for variables', () => {
      const columns = getVariableColumns({ ...tableProps, tabId: INSTANCE_TABS_TYPES.VARIABLES });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for incidents', () => {
      const columns = getIncidentsColumns({ ...tableProps, tabId: INSTANCE_TABS_TYPES.INCIDENTS });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for called processes', () => {
      const columns = getCalledProcessesColumns({ ...tableProps, tabId: INSTANCE_TABS_TYPES.CALLED_PROCESS });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for job definitions', () => {
      const columns = getJobDefinitionsColumns({ ...tableProps, tabId: INSTANCE_TABS_TYPES.JOB_DEFINITIONS });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for external tasks', () => {
      const columns = getExternalTasksColumns({ ...tableProps, tabId: INSTANCE_TABS_TYPES.EXTERNAL_TASKS });
      expect(columns).toMatchSnapshot();
    });
  });
});
