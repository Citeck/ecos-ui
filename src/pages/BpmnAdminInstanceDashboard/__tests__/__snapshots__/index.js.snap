// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`  should render MetaInfo component 1`] = `
<div
  class="dashlet"
>
  <div
    class="ecos-well ecos-panel dashlet"
  >
    <div
      class="ecos-panel__head dashlet__header-wrapper"
    >
      <div
        class="dashlet__header"
      >
        <span
          class="dashlet__caption"
        >
          <i
            class="icon dashlet__header-collapser icon-small-up"
          />
          instance-admin.info-widget.title
        </span>
        <span
          class="ecos-badge ecos-badge_info ecos-badge_large"
        />
        <div
          class="dashlet__header-actions"
        />
      </div>
    </div>
    <div
      class="ecos-panel__body dashlet__body"
    >
      <div
        class="dashlet__body-content"
      >
        <div
          class="bpmn-admin-info"
        >
          <div
            class="bpmn-admin-info__main"
          >
            <div
              class="bpmn-admin-info__main-item"
            >
              <div
                class="ecos-panel-title ecos-panel-title_color-gray ecos-panel-title_narrow"
              >
                instance-admin.info-widget.document
              </div>
              <button
                class="ecos-btn ecos-btn_disabled ecos-btn_blue"
                disabled=""
              >
                <div
                  class="ecos-btn__content"
                >
                  instance-admin.info-widget.not-defined
                  <i
                    class="icon-small-right"
                  />
                </div>
              </button>
            </div>
            <div
              class="bpmn-admin-info__main-item"
            >
              <div
                class="ecos-panel-title ecos-panel-title_color-gray ecos-panel-title_narrow"
              >
                instance-admin.info-widget.actions-title
              </div>
              <div
                class="dropdown"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="true"
                  class=""
                >
                  <button
                    class="ecos-btn bpmn-admin-info__actions ecos-btn_blue"
                  >
                    <div
                      class="ecos-btn__content"
                    >
                      instance-admin.info-widget.choose-action
                      <i
                        class="icon-small-down"
                      />
                    </div>
                  </button>
                </div>
                <div
                  aria-hidden="true"
                  class="dropdown-menu dropdown-menu-right"
                  container="body"
                  role="menu"
                  tabindex="-1"
                >
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    migration-modal.dropdown-title
                  </button>
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    <a
                      class="bpmn-admin-info__dropdown-link"
                      href="/v2/bpmn-migration?recordRef=undefined&fromVersion=undefined"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      migration-versions.dropdown-title
                    </a>
                  </button>
                </div>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                />
              </div>
            </div>
            <div
              class="bpmn-admin-info__main-item"
            >
              <div
                class="ecos-panel-title ecos-panel-title_color-gray ecos-panel-title_narrow"
              >
                instance-admin.info-widget.definition-ref
              </div>
              <button
                class="ecos-btn ecos-btn_disabled ecos-btn_blue"
                disabled=""
              >
                <div
                  class="ecos-btn__content"
                >
                  instance-admin.info-widget.not-defined
                  <i
                    class="icon-small-right"
                  />
                </div>
              </button>
            </div>
          </div>
          <div
            class="bpmn-admin-info__secondary"
          >
            <span>
              instance-admin.info-widget.version: instance-admin.info-widget.not-defined
            </span>
            <span>
              instance-admin.info-widget.start-time: instance-admin.info-widget.not-defined
            </span>
          </div>
        </div>
      </div>
      <div
        class="dashlet__body-indent dashlet__body-indent_bottom"
      />
    </div>
  </div>
  <div />
</div>
`;

exports[`  should render corrent columns for called processes 1`] = `
Array [
  Object {
    "attribute": "calledProcessDisp",
    "dataField": "calledProcessDisp",
    "newFormatter": Object {
      "config": Object {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "text": "Called instance",
  },
  Object {
    "attribute": "bpmnDefEngineDisp",
    "dataField": "bpmnDefEngineDisp",
    "newFormatter": Object {
      "config": Object {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "text": "Process definition",
  },
  Object {
    "attribute": "callActivityId",
    "dataField": "callActivityId",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Element",
  },
  Object {
    "attribute": "id",
    "dataField": "id",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "State",
  },
]
`;

exports[`  should render corrent columns for external tasks 1`] = `
Array [
  Object {
    "attribute": "id",
    "dataField": "dispId",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "predicates": Array [
      Object {
        "label": "predicate.eq",
        "needValue": true,
        "value": "eq",
      },
    ],
    "searchable": true,
    "searchableByText": true,
    "sortable": true,
    "text": "ID",
  },
  Object {
    "attribute": "activityId",
    "dataField": "activityId",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "predicates": Array [
      Object {
        "label": "predicate.eq",
        "needValue": true,
        "value": "eq",
      },
    ],
    "searchable": true,
    "searchableByText": true,
    "text": "Element",
  },
  Object {
    "attribute": "retries",
    "dataField": "retries",
    "text": "Retries",
  },
  Object {
    "attribute": "workerId",
    "dataField": "workerId",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "predicates": Array [
      Object {
        "label": "predicate.eq",
        "needValue": true,
        "value": "eq",
      },
    ],
    "searchable": true,
    "searchableByText": true,
    "text": "Handler's ID",
  },
  Object {
    "attribute": "lockExpirationTime",
    "dataField": "lockExpirationTime",
    "newFormatter": Object {
      "config": Object {
        "format": "LLL",
      },
      "type": "date",
    },
    "text": "Expiration time",
  },
  Object {
    "attribute": "topicName",
    "dataField": "topic",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "predicates": Array [
      Object {
        "label": "predicate.eq",
        "needValue": true,
        "value": "eq",
      },
    ],
    "searchable": true,
    "searchableByText": true,
    "text": "Topic",
  },
  Object {
    "attribute": "priority",
    "dataField": "priority",
    "sortable": true,
    "text": "Priority",
  },
]
`;

exports[`  should render corrent columns for incidents 1`] = `
Array [
  Object {
    "attribute": "message",
    "dataField": "message",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "searchable": true,
    "searchableByText": true,
    "text": "Message",
  },
  Object {
    "attribute": "processInstance",
    "dataField": "processInstance",
    "newEditor": Object {
      "config": Object {},
      "type": "text",
    },
    "newFormatter": Object {
      "config": Object {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "text": "Process instance",
  },
  Object {
    "attribute": "created",
    "dataField": "created",
    "newFormatter": Object {
      "config": Object {
        "format": "LLL",
      },
      "type": "date",
    },
    "text": "Created",
  },
  Object {
    "attribute": "activityId",
    "dataField": "activityId",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Element",
  },
  Object {
    "attribute": "activityId",
    "dataField": "activityId",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Failed element",
  },
  Object {
    "attribute": "incidentType",
    "dataField": "incidentType",
    "text": "Type",
  },
  Object {
    "attribute": "note",
    "dataField": "note",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Note",
  },
  Object {
    "attribute": "id",
    "dataField": "id",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Actions",
  },
]
`;

exports[`  should render corrent columns for job definitions 1`] = `
Array [
  Object {
    "attribute": "id",
    "dataField": "dispId",
    "sortable": true,
    "text": "ID",
  },
  Object {
    "attribute": "created",
    "dataField": "created",
    "newFormatter": Object {
      "config": Object {
        "format": "LLL",
      },
      "type": "date",
    },
    "text": "Created",
  },
  Object {
    "attribute": "dueDate",
    "dataField": "dueDate",
    "newFormatter": Object {
      "config": Object {
        "format": "LLL",
      },
      "type": "date",
    },
    "sortable": true,
    "text": "Due date",
  },
  Object {
    "attribute": "retries",
    "dataField": "retries",
    "sortable": true,
    "text": "Retries",
  },
  Object {
    "attribute": "activityId",
    "dataField": "activityId",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Element",
  },
  Object {
    "attribute": "failedActivityId",
    "dataField": "failedActivityId",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Failed element",
  },
  Object {
    "attribute": "configuration",
    "dataField": "configuration",
    "text": "Configuration",
  },
  Object {
    "attribute": "id",
    "dataField": "id",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Actions",
  },
]
`;

exports[`  should render corrent columns for variables 1`] = `
Array [
  Object {
    "attribute": "name",
    "dataField": "name",
    "sortable": true,
    "text": "Name",
  },
  Object {
    "attribute": "type",
    "dataField": "type",
    "text": "Type",
  },
  Object {
    "attribute": "value",
    "dataField": "value",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Value",
  },
  Object {
    "attribute": "scope",
    "dataField": "scope",
    "disableSelect": true,
    "newEditor": Object {
      "config": Object {
        "journalId": "bpmn-process-def",
        "multiple": false,
      },
      "type": "journal",
    },
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "searchable": true,
    "searchableByText": true,
    "text": "Scope",
  },
  Object {
    "attribute": "id",
    "classes": "common-table-column__small",
    "dataField": "id",
    "newFormatter": Object {
      "config": Object {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Actions",
  },
]
`;
